odoo/addons/base/models/ir_model.py

```diff
@@ -2018,12 +2018,14 @@ def _process_end(self, modules):
                     _logger.info('Deleting %s@%s (%s)', res_id, model, xmlid)
                     record = self.env[model].browse(res_id)
                     if record.exists():
-                        if record._name == 'ir.model.fields':
+                        if record._name == 'ir.model.fields' and not xmlid.startswith('test_'):
                             _logger.warning(
                                 "Deleting field %s.%s (hint: fields should be"
                                 " explicitly removed by an upgrade script)",
                                 record.model, record.name,
                             )
+                        elif record._name == 'ir.model' and not xmlid.startswith('test_'):
```

Comment:
I would maybe chose a more generic name than drop_table_warning, like in_test_module. It would also be a default False context key. 
I think we could also always give the context key every time, not only for 'ir.model'.

Reply:
> I would maybe chose a more generic name than drop_table_warning, like in_test_module.

There is only one case.  I don't see the point of a generic name.

> It would also be a default False context key.

Very bad practice, because it inevitably introduces double negations in your code, which makes it less readable.

> I think we could also always give the context key every time, not only for 'ir.model'.

Why?